src/app/subnets/views/SubnetDetails/StaticDHCPLease/ReserveDHCPLease/ReserveDHCPLease.tsx

```diff
@@ -21,28 +21,63 @@ import {
   isIpInSubnet,
 } from "@/app/utils/subnetIpRange";
 
-type Props = Pick<SubnetActionProps, "subnetId" | "setSidePanelContent">;
+const MAX_COMMENT_LENGTH = 255;
+
+type Props = Pick<
+  SubnetActionProps,
+  "subnetId" | "setSidePanelContent" | "reservedIpId"
+>;
 
 type FormValues = {
   ip_address: string;
   mac_address: string;
   comment: string;
 };
 
-const ReserveDHCPLease = ({ subnetId, setSidePanelContent }: Props) => {
+const ReserveDHCPLease = ({
+  subnetId,
+  setSidePanelContent,
+  reservedIpId,
+}: Props) => {
   const subnet = useSelector((state: RootState) =>
     subnetSelectors.getById(state, subnetId)
   );
+  const reservedIp = useSelector((state: RootState) =>
+    reservedIpSelectors.getById(state, reservedIpId)
+  );
   const subnetLoading = useSelector(subnetSelectors.loading);
   const reservedIpLoading = useSelector(reservedIpSelectors.loading);
   const errors = useSelector(reservedIpSelectors.errors);
   const saving = useSelector(reservedIpSelectors.saving);
   const saved = useSelector(reservedIpSelectors.saved);
+  const dispatch = useDispatch();
+  const cleanup = useCallback(() => reservedIpActions.cleanup(), []);
 
   const loading = subnetLoading || reservedIpLoading;
+  const isEditing = !!reservedIpId;
+
+  const getInitialValues = () => {
```

Comment:
In order for this to work as expected you'd need to enable `enableReinitialize`, otherwise if any of the values change after initial render, initial formik state won't be updated.